tfhe/src/shortint/engine/server_side/mod.rs

```diff
@@ -290,6 +290,18 @@ impl ShortintEngine {
         })
     }
 
+    pub(crate) fn generate_msg_lookup_table<F>(
+        &mut self,
+        server_key: &ServerKey,
+        f: F,
+        modulus: MessageModulus,
+    ) -> EngineResult<LookupTableOwned>
+    where
+        F: Fn(u64) -> u64,
+    {
+        Self::generate_lookup_table_with_engine(server_key, |x| f(x % modulus.0 as u64))
+    }
```

Comment:
I wonder, if this applies modulus to the input, would it be coherent for it to also apply the modulus on the output ?

Reply:
I guess if we want to ignore the carry bits of the input, we also want to ignore the ones of the output
So IMO it makes sense to apply the same modulus on the ouput here

Reply:
Done